### PR TITLE
Make CoreDNS default in kubeup and update CoreDNS version/manifest in kubeup and kubeadm

### PR DESCRIPTION
**Pull request by chrisohaver**
**What this PR does / why we need it**:

Makes CoreDNS default for kube-up 
Updates manifest and CoreDNS version for kube-up and kubeadm.

KEP: https://github.com/kubernetes/community/blob/master/keps/sig-network/0012-20180518-coredns-default-proposal.md
Feature: kubernetes/features#566

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

This version of CoreDNS (1.2.4) contains improvements that reduce memory usage at scale.  Based on the 500 node and 2000 node e2e scale test results, we have linearly projected that memory use in the 5000 node scale test should be less than the prescribed resource limit.  But since the 5000 node scale test is not available as a pre-submit, we cannot confirm until this PR is merged.

**Release note**:

```release-note
CoreDNS is now the default DNS server in kube-up deployments. 
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/addons/dns/manifests.go [hunk 1]
```diff
@@ -331,6 +331,12 @@ rules:
   verbs:
   - list
   - watch
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/dns/manifests.go]: i'm pretty sure these should be under `rules:`.
http://people.redhat.com/jrivera/openshift-docs_preview/openshift-online/glusterfs-review/rest_api/apis-rbac.authorization.k8s.io/v1.ClusterRole.html#object-schema

i guess this applies to the rest of the files that are changed from the diff.

**Review by neolit123 — COMMENTED** [review]
@chrisohaver thanks for the update to coredns 1.2.3.
added one minor comment.

File: cmd/kubeadm/app/phases/addons/dns/manifests.go (hunk 1 above)
Comment (shashidharatd) [reply cmd/kubeadm/app/phases/addons/dns/manifests.go]: @neolit123, they are under `rules:` as one more list item. Am i missing something here?

Comment (neolit123) [reply cmd/kubeadm/app/phases/addons/dns/manifests.go]: never mind, i got confused by the indentation.

**Comment by chrisohaver** [comment]
We've identified a [bug](https://github.com/coredns/coredns/issues/2204) in CoreDNS 1.2.3 that makes it invalid for kubernetes deployment.
Closing this issue.  Will re-open, or re-sibmit a new PR with new version when the CoreDNS issue is resolved and new version cut.

**Comment by chrisohaver** [comment]
Latest commit is intended to run the latest coredns release (1.2.4) through e2e presubmit tests prior to pushing the coredns image to gcr.io. If all is OK, we will kick off the push to gcr.io, and I'll update the image locations in the manifests here (from `coredns/coredns` to `gcr.io/coredns`).

**Comment by chrisohaver** [comment]
I suspect the e2e test nodes don't have access staging-k8s.gcr.io, hence failure to pull images, and subsequent test failure.  Will test now on k8s.gcr.io, since latest version appears to be already promoted.

**Comment by chrisohaver** [comment]
I've squashed the commits.  Commit 0cfb4bbe2cb2e3d64ec7f281bf86c439da611f73 is now pointing to the k8s.gcr.io repo.

**Comment by neolit123** [comment]
> Will test now on k8s.gcr.io, since latest version appears to be already promoted.

yes, the image up.

> Make CoreDNS default and update version

there have been quite a bit of confusion about this part over the last couple of cycles - "default in what?"
i think the first sentence in the description summarizes it better:
> Makes CoreDNS default for kube-up

**Comment by AishSundar** [comment]
@jberkus and @mortent as FYI

**Comment by shashidharatd** [comment]
How about merging this pr to other release branches? (v1.11 and v1.12). Does this pr qualify to be back ported to previous releases?

**Comment by chrisohaver** [comment]
> How about merging this pr to other release branches? (v1.11 and v1.12). Does this pr qualify to be back ported to previous releases?

I don't believe we can port the entire change back, i.e. Retroactively making CoreDNS the default cluster DNS for kubeup installations.

But we could push newer default versions/manifests of coredns into kubedns/kubeadm of older versions.  I'd think this would be reserved for security fixes and major bug fixes, although I'm not personally familiar with the policy for support on past versions of k8s.  Question for the sig-lifecycle team I think?

**Comment by neolit123** [comment]
> How about merging this pr to other release branches? (v1.11 and v1.12). Does this pr qualify to be back ported to previous releases?

nope, not really.

**Comment by D-PENG** [comment]
good

**Pull request merged**